urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/ia/evaluacion/EvaluacionEstrategica.java

```diff
@@ -0,0 +1,223 @@
+		int totalCasillasCentrales = 4;
+		int casillasCentralesJugador=0;
+		int casillasCentralesRival=0;
+		EstadoCasilla colorRival = colorJugador == EstadoCasilla.BLACK ? EstadoCasilla.WHITE : EstadoCasilla.BLACK;
+		List<Point> centrales = new ArrayList<Point>(4);
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2)-1, (int)(Tablero.TABLERO_LARGO/2)-1));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2)-1, (int)(Tablero.TABLERO_LARGO/2)));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2), (int)(Tablero.TABLERO_LARGO/2)-1));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2), (int)(Tablero.TABLERO_LARGO/2)));
+		ListIterator<Point> iterador = centrales.listIterator();
+		while( iterador.hasNext() ) {
+		       Point punto = (Point) iterador.next();
+				if (tablero.obtenerTablero().get(punto) == colorJugador){
+					casillasCentralesJugador++;
+				}else if (tablero.obtenerTablero().get(punto) == colorRival) {
+					casillasCentralesRival++;
+				}
+		}
+		
+		/*System.out.println(colorJugador +" = " + casillasCentralesJugador);
+		System.out.println(colorRival +" = " + casillasCentralesRival);
+		System.out.println((double)(casillasCentralesJugador-casillasCentralesRival)/(double)totalCasillasCentrales);*/
+		return (double)(casillasCentralesJugador-casillasCentralesRival)/(double)totalCasillasCentrales;
+	}
+	
+
+	private List<Point> detectarEsquinas(Tablero tablero, TipoTablero tipoTablero) {
+		List<Point> esquinas = null;
+		if (tipoTablero == TipoTablero.CLASICO) {
+			esquinas = new ArrayList<Point>(4);
+			esquinas.add(new Point(1, 1));
+			esquinas.add(new Point(Tablero.TABLERO_LARGO-2, 1));
+			esquinas.add(new Point(1, Tablero.TABLERO_ANCHO-2));
+			esquinas.add(new Point(Tablero.TABLERO_LARGO-2, Tablero.TABLERO_ANCHO-2));
+		} if (tipoTablero == TipoTablero.OCTOGONAL) {
+			/*esquinas = new ArrayList<Point>(8);
+			esquinas.add(new Point( 0, (int)(Tablero.TABLERO_LARGO/2)-1) );
+			esquinas.add(new Point( 0, (int)(Tablero.TABLERO_LARGO/2))   );
+			esquinas.add(new Point( (int)((Tablero.TABLERO_ANCHO/2)-1), 0) );
+			esquinas.add(new Point( (int)((Tablero.TABLERO_ANCHO/2)), 0) );
+			esquinas.add(new Point( (int)((Tablero.TABLERO_ANCHO/2)-1), Tablero.TABLERO_LARGO-1) );
+			esquinas.add(new Point( (int)((Tablero.TABLERO_ANCHO/2)), Tablero.TABLERO_LARGO-1) );
+			esquinas.add(new Point( Tablero.TABLERO_LARGO-1, (int)(Tablero.TABLERO_LARGO/2)-1) );
+			esquinas.add(new Point( Tablero.TABLERO_LARGO-1, (int)(Tablero.TABLERO_LARGO/2))   );*/
+			esquinas = new ArrayList<Point>();
+			Map<Point, EstadoCasilla> tableroMap = tablero.obtenerTablero();
+			for (int x = 0; x < Tablero.TABLERO_LARGO; x++) {
+				for (int y = 0; y < Tablero.TABLERO_ANCHO; y++) {
+					if(tableroMap.get(new Point(x,y)) != EstadoCasilla.WALL){
+						if(esEsquina(new Point(x,y),tablero)){
+							esquinas.add(new Point(x,y));
+						}
+					}
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 160:168, 172:180.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.